### DISCUSSION TITLE
etcd-operator adoption

### DISCUSSION THREAD

**Discussion by kvaps** [gravitational/teleport#40407] [Show and tell]
Hey, as far I know Teleport requires etcd for the cluster state storage, but you don't provide any infromation how to install and mange it.

I'd like to introduce that we've initiated a group effort to create a generic, multi-purpose etcd-operator. Currently, the project is in its early development phase, so there's a good opportunity for you to make influence on the project.

https://github.com/aenix-io/etcd-operator

Our development process is open, and we are in [discussions](https://github.com/kubernetes/community/pull/7796) with sig-etcd about the possibility of making this the official version under Kubernetes-SIGs. Our goal is to bring together all potential adopters.

We would be pleased if you could present your official stance on this initiative and respond to the following questions:
- Whether your project requires an etcd-operator at all.
- Would you like to adopt and start using it once a stable version is released?
- How important is it to you that the project continues to develop under the control of the official sig-etcd?
- Would you be interested in joining the development and discussions to help address architectural challenges?

We welcome any other ideas and suggestions you have regarding this initiative.

**Comment by zmb3** [comment]
Hi @kvaps - I've converted this to a discussion which is a better forum for feedback, ideas, and suggestions. (We prefer to use GitHub issues for more actionable work like development tasks and bug reports).

To add some color:

> as far I know Teleport requires etcd for the cluster state storage

etcd is one of our supported backends for cluster state, but it is not required. The majority of Teleport clusters we are aware of run on AWS with the DynamoDB backend, but etcd is popular in large/complex self-hosted deployments.

You're correct to point out that we don't provide much guidance as to how to install or operate etcd. etcd is notoriously difficult to operate at scale, and so there's a bit of an assumption on our part that if you're advanced enough to run your own etcd deployment that you probably have your own preferred way of operating it.

It sounds like there is a desire for an official etcd operator but it is unclear from where that will emerge. We'll definitely keep an eye out!
